bin/zowe-setup-certificates.sh

```diff
@@ -0,0 +1,180 @@
+#                       that will be also used to secure newly generated keystores for API Mediation.
+# - ZOWE_USER_ID - zowe user id to set up ownership of the generated certificates
+
+
+# Set up logging
+LOG_DIR=${HOME}/zowe_certificate_setup_log
+# Make the log directory if needed - first time through - subsequent installs create new .log files
+if [[ ! -d $LOG_DIR ]]; then
+    mkdir -p $LOG_DIR
+    chmod a+rwx $LOG_DIR 
+fi
+
+export LOG_FILE="certificate_config_`date +%Y-%m-%d-%H-%M-%S`.log"
+LOG_FILE=$LOG_DIR/$LOG_FILE
+touch $LOG_FILE
+chmod a+rw $LOG_FILE
+
+echo "<zowe-setup-certificates.sh>" >> $LOG_FILE
+
+# process input parameters. 
+while getopts "p:" opt; do
+  case $opt in
+    p) CERTIFICATES_CONFIG_FILE=$OPTARG;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      exit 1
+      ;;
+  esac
+done
+shift "$(($OPTIND-1))"
+
+export ZOWE_ROOT_DIR=$(cd $(dirname $0)/../;pwd)
+
+# Load default values
+DEFAULT_CERTIFICATES_CONFIG_FILE=${ZOWE_ROOT_DIR}/bin/zowe-setup-certificates.env
+echo "Loading default variables from ${DEFAULT_CERTIFICATES_CONFIG_FILE} file."
+. ${DEFAULT_CERTIFICATES_CONFIG_FILE}
+
+if [[ -z ${CERTIFICATES_CONFIG_FILE} ]]
+then
+  echo "-p parameter not set. Using default ${DEFAULT_CERTIFICATES_CONFIG_FILE} file instead."
+else
+  if [[ -f ${CERTIFICATES_CONFIG_FILE} ]]
+  then
+    echo "Loading ${CERTIFICATES_CONFIG_FILE} file and overriding default variables."
+    # Load custom values
+    . ${CERTIFICATES_CONFIG_FILE}  
+  else
+    echo "${CERTIFICATES_CONFIG_FILE} file does not exist."
+    exit 1
+  fi
+fi
+
+. ${ZOWE_ROOT_DIR}/bin/zowe-init.sh 
+. ${ZOWE_ROOT_DIR}/scripts/utils/configure-java.sh
+
+# If hostname was not provided then use a default value.
+if [[ -z ${HOSTNAME} ]]
+then
+  HOSTNAME=${ZOWE_EXPLORER_HOST}
+fi
+
+# If ipaddress was not provided then use a default value.
+if [[ -z ${IPADDRESS} ]]
+then
+  IPADDRESS=${ZOWE_IP_ADDRESS}
+fi
+
+ZOWE_CERT_ENV_NAME=zowe-certificates.env
+LOCAL_KEYSTORE_SUBDIR=local_ca
+
+# create keystore directories
+if [ ! -d ${KEYSTORE_DIRECTORY}/${LOCAL_KEYSTORE_SUBDIR} ]; then
+  if ! mkdir -p ${KEYSTORE_DIRECTORY}/${LOCAL_KEYSTORE_SUBDIR}; then 
+    echo "Unable to create ${KEYSTORE_DIRECTORY}/${LOCAL_KEYSTORE_SUBDIR} directory."
+    exit 1;
+  fi
+fi
+
+if [ ! -d ${KEYSTORE_DIRECTORY}/${KEYSTORE_ALIAS} ]; then
+  if ! mkdir -p ${KEYSTORE_DIRECTORY}/${KEYSTORE_ALIAS}; then
+    echo "Unable to create ${KEYSTORE_DIRECTORY}/${KEYSTORE_ALIAS} directory."
+    exit 1;
+  fi
+fi
+
+# set up parameters fot apiml_cm.sh script
+KEYSTORE_PREFIX="${KEYSTORE_DIRECTORY}/${KEYSTORE_ALIAS}/${KEYSTORE_ALIAS}.keystore"
+TRUSTSTORE_PREFIX="${KEYSTORE_DIRECTORY}/${KEYSTORE_ALIAS}/${KEYSTORE_ALIAS}.truststore"
+EXTERNAL_CA_PREFIX=${KEYSTORE_DIRECTORY}/${LOCAL_KEYSTORE_SUBDIR}/extca
+LOCAL_CA_PREFIX=${KEYSTORE_DIRECTORY}/${LOCAL_KEYSTORE_SUBDIR}/localca
+SAN="SAN=dns:${HOSTNAME},ip:${IPADDRESS},dns:localhost.localdomain,dns:localhost,ip:127.0.0.1"
```

Comment:
With the above change in place this would become~
`SAN="SAN=dns:${ZOWE_EXPLORER_HOST},ip:${ZOWE_IP_ADDRESS},dns:localhost.localdomain,dns:localhost,ip:127.0.0.1"`